### PR TITLE
[FIXED] JetStream: do not automatically set MaxAckPending

### PR DESCRIPTION
**Pull request by kozlovic**
We were still setting the MaxAckPending value to the capacity
of the go channel for synchronous or channel subscriptions. The
issue is that the server may have a limit imposed on this value
which would prevent creation of synchronous subscriptions without
having to create a connection with a lower SyncQueueLen...

Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

### CODE REVIEW

File: js.go [hunk 1]
```diff
@@ -1465,12 +1465,10 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync,
 		if cfg.ReplayPolicy == replayPolicyNotSet {
 			cfg.ReplayPolicy = ReplayInstantPolicy
 		}
+		// Note: Do not set MaxAckPending if not set by the user. The server may
+		// have limits that are lower than the sync subscription go channel size,
+		// so don't use the cap size as the "default" MaxAckPending.
```
[the remark this answers is not in the document]
Comment (derekcollison) [reply js.go:1470]: And if they set it higher, that is ok vs the server default.

**Review by wallyqs — APPROVED** [review]
LGTM, default sync queue len is then 64K for the channels and since v2.8.0 default max ack pending is 1000 so channel would be lower than server defaults most of the time then.

File: js.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (wallyqs) [reply js.go:1470]: Maybe could add an check that SynqQueueLen or channel size should not be less than 1000 server ack pending default when creating the JetStream context or a channel based subscriber.

**Comment by kozlovic** [comment]
@wallyqs I will revisit this PR after chatting with Derek. It may be enough that the server returns the actual limit value in the error message and have then the user call subscribe with `nats.MaxAckPending`. If the value is expected to be higher than the sync queue len, then user can use the SyncQueueLen() option on Connect.

**Comment by kozlovic** [comment]
@wallyqs Closing this PR and will re-open a new one with updates to NATS Server to latest (once I merge the PR that indicates what is the limit being exceeded). So there is no need to modify the js's subscribe() call. But the PR will still fix flappers and changes needed to work with newer server.

**Pull request closed without merging**